### PR TITLE
Introduce tests on mobile screen sizes

### PR DESCRIPTION
**Pull request by KKoukiou**
Tested locally with: TEST_MOBILE=true TEST_OS=fedora-34 test/verify/check-networkmanager-basic -st TestNetworkingBasic.testBasic

It created a: TestNetworkingBasic-testBasic-network-main-mobile-pixels.png file as expected.

### CODE REVIEW

File: test/common/cdp.py [hunk 1]
```diff
@@ -170,12 +171,15 @@ def browser_cmd(self, cdp_port, env):
         if not exe:
             raise SystemError(self.browser + " is not installed")
 
+        ws = "1920x1200"
```
Comment (marusak) [inline test/common/cdp.py:174]: on line 204 we set size of firefox browser. I would say that setting `TEST_MOBILE` env should work for both (even if we don't have context for it right now)

File: test/run [hunk 2]
```diff
@@ -14,6 +14,7 @@ case $TEST_SCENARIO in
     verify|devel|firefox|firefox-devel)
         [ "${TEST_SCENARIO}" = "${TEST_SCENARIO%%devel}" ] || export NODE_ENV=development
         [ "${TEST_SCENARIO}" = "${TEST_SCENARIO##firefox}" ] || export TEST_BROWSER=firefox
+        [ "${TEST_SCENARIO}" = "${TEST_SCENARIO##mobile}" ] || export TEST_MOBILE=""
```
Comment (martinpitt) [inline test/run:17]: This needs to be set to an actual value.

**Review by martinpitt — CHANGES_REQUESTED** [review]

**Comment by martinpitt** [comment]
So this PR would mean that we would immediately need to provide -mobile versions of all 14 pixel tests, right? That seems fine to me.

**Comment by KKoukiou** [comment]
> So this PR would mean that we would immediately need to provide -mobile versions of all 14 pixel tests, right? That seems fine to me.

Correct.

File: test/common/cdp.py (hunk 1 above)
Comment (KKoukiou) [reply test/common/cdp.py:174]: Fixed.

File: test/run (hunk 2 above)
Comment (KKoukiou) [reply test/run]: Fixed.

**Comment by KKoukiou** [comment]
@marusak @martinpitt What do you think about increasing the vertical size (a lot) in order that we get more content in each mobile screenshot? 

Now they look like:
TestNetworkingBasic-testBasic-network-main-mobile-pixels


Increasing the vertical size only want make us lose something from the expected behavior of the CSS etc so I think it's a win - win.

**Comment by martinpitt** [comment]
@KKoukiou : Agreed -- I think all our CSS only looks at the width. Talller screens are more useful to visually debug test failures, and we can still imagine how a real screen would look like.

File: test/common/cdp.py [hunk 3]
```diff
@@ -197,7 +201,10 @@ def browser_cmd(self, cdp_port, env):
             with open(os.path.join(profile, "handlers.json"), "w") as f:
                 f.write('{"defaultHandlersVersion":{"en-US":4},"mimeTypes":{"application/xz":{"action":0,"extensions":["xz"]}}}')
 
-            cmd = [exe, "-P", "blank", "--window-size=1920,1200", "--remote-debugging-port=%i" % cdp_port, "--no-remote", "localhost"]
+            ws = "1920,1200"
+            if self.mobile:
+                ws = "414,736"  # iPhone 6/7/8 Plus
```
Comment (martinpitt) [inline test/common/cdp.py:206]: Please don't repeat the numbers -- can you set self.width and self.height in the constructor and just assemble the strings here? (comma vs 'x')

File: test/verify/check-superuser [hunk 4]
```diff
@@ -245,6 +261,23 @@ class TestSuperuserOldShell(MachineCase):
         "machine2": {"address": "10.111.113.2/20", "image": "centos-7"},
     }
 
+    def checkSuperuserIndicator(self, access):
```
Comment (martinpitt) [inline test/verify/check-superuser:264]: These two helpers seem common enough to move to the `Browser` class in testlib.py, as they would often even apply to third-party projects. Even now they are already copied four times.

**Review by martinpitt — APPROVED** [review]
Very cool! This is conflict prone and will get some follow-ups anyway, so let's defer the helper factorization and get this in ASAP. (Just waiting for packit)

Thanks a lot for working on this! :heart:

File: test/verify/check-superuser (hunk 4 above)
Comment (KKoukiou) [reply test/verify/check-superuser:264]: Sure- will move it right after.

**Comment by martinpitt** [comment]
packit failures are due to non-rebased branch. The test does run in general.

**Pull request merged**